From 6788d44e9326ee288aa4e6650a35e63a9ba261d4 Mon Sep 17 00:00:00 2001
From: Ronald Brown <rbrown4014@yahoo.com>
Date: Tue, 2 Dec 2025 19:02:50 +0100
Subject: [PATCH] linux: sensors: illminance: Fix failure to open file
 resulting in success code being sent.

The way the sensor init function returns >=0 is a success code, as on success it sets initial illuminance sensor reading which is a value between 0 and...
by setting it to 0 on failure to open file, sets an invalid path, then starts poller on that path, anf breaks search loop.
---
 input/common/linux_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/input/common/linux_common.c b/input/common/linux_common.c
index 8d2dfed76e1d..0f4207898ddc 100644
--- a/input/common/linux_common.c
+++ b/input/common/linux_common.c
@@ -320,7 +320,7 @@ static double linux_read_illuminance_sensor(const linux_illuminance_sensor_t *se
    if (!in_illuminance_input)
    {
       RARCH_ERR("Failed to open \"%s\".\n", sensor->path);
-      return 0.0;
+      return -1.0;
    }
 
    /* Read the illuminance value from the file. If that fails... */
